engine/src/main/java/org/apache/apex/engine/events/grouping/GroupingRequest.java

```diff
@@ -129,6 +129,18 @@ public void addAffectedContainer(String containerId)
   }
 
   /**
+   * Checks if request is processed
+   * @return isProcessed
+   */
+  public boolean isProcessed()
+  {
+    if (getOperatorsToDeploy().isEmpty() && getOperatorsToUndeploy().isEmpty()) {
```

Comment:
Please see my original comment. The result of isProcessed() is the value of operatorsToDeploy.isEmpty() && operatorsToUndeploy.isEmpty(). There is no need to extra if